src/views/configuration/ImportConfiguration.vue

```diff
@@ -1,28 +1,108 @@
 <template>
   <v-container>
+    <v-card class="mb-4">
+      <v-card-title>Search Location</v-card-title>
+      <v-card-text>
+        <div class="d-flex align-center">
+          <span class="mr-4">
+            <strong>Current folder:</strong>
+            {{ currentFolderName || "Loading..." }}
+          </span>
+          <girder-location-chooser
+            v-model="selectedFolder"
+            :breadcrumb="true"
+            title="Select a folder to search for collections"
+          />
+        </div>
+      </v-card-text>
+    </v-card>
     <configuration-select
       @submit="submit"
       @cancel="cancel"
       :title="`Add dataset ${datasetName} to one or several existing collections`"
+      :folderId="folderId"
     />
   </v-container>
 </template>
 <script lang="ts">
-import { Vue, Component } from "vue-property-decorator";
+import { Vue, Component, Watch } from "vue-property-decorator";
 import store from "@/store";
+import girderResources from "@/store/girderResources";
 import ConfigurationSelect from "@/components/ConfigurationSelect.vue";
+import GirderLocationChooser from "@/components/GirderLocationChooser.vue";
 import { IDatasetConfiguration } from "@/store/model";
+import { IGirderSelectAble } from "@/girder";
 
 @Component({
-  components: { ConfigurationSelect },
+  components: { ConfigurationSelect, GirderLocationChooser },
 })
 export default class ImportConfiguration extends Vue {
   readonly store = store;
+  readonly girderResources = girderResources;
+
+  selectedFolder: IGirderSelectAble | null = null;
+  currentFolderName: string = "";
 
   get datasetName() {
     return this.store.dataset?.name || "";
   }
 
+  get folderId(): string | undefined {
+    if (this.selectedFolder && this.selectedFolder._modelType === "folder") {
+      return this.selectedFolder._id;
+    }
+    // Fallback to route query if folder not yet loaded
+    const folderId = this.$route.query.folderId;
+    if (Array.isArray(folderId)) {
+      return folderId[0] || undefined;
+    }
+    if (folderId === null || folderId === undefined) {
```

Comment:
This can be replaced with `if(folderId ==null)`

Reply:
Good point, thanks!